dtfs-central-api/api-tests/v1/deals/get-deal.api-test.js

```diff
@@ -83,5 +92,69 @@ describe('/v1/portal/deals', () => {
         expect(body.deal.loanTransactions.items).toEqual([loan1, loan2]);
       });
     });
+
+    describe('when the deal is cancelled in tfm', () => {
+      beforeEach(async () => {
+        process.env.FF_TFM_DEAL_CANCELLATION_ENABLED = 'true';
+
+        await testApi
+          .put({
+            dealType: DEALS.DEAL_TYPE.BSS_EWCS,
+            dealId,
+            auditDetails: generatePortalAuditDetails(MOCK_PORTAL_USER._id),
+          })
+          .to('/v1/tfm/deals/submit');
+
+        await testApi
+          .put({
+            dealUpdate: {
+              tfm: {
+                dateReceived: '23-09-2024',
+                dateReceivedTimestamp: 1727085149,
+                parties: {},
+                activities: [],
+                product: DEALS.DEAL_TYPE.BSS_EWCS,
+                stage: TFM_DEAL_STAGE.CONFIRMED,
+                exporterCreditRating: 'Acceptable (B+)',
+                lastUpdated: 1727085149571,
+                lossGivenDefault: 50,
+                probabilityOfDefault: 12,
+              },
+            },
+            auditDetails: generatePortalAuditDetails(MOCK_PORTAL_USER._id),
+          })
+          .to(`/v1/tfm/deals/${dealId}`);
+
+        const tfmDealCancellation = { reason: '', effectiveFrom: new Date().valueOf(), bankRequestDate: new Date().valueOf() };
+
+        await testApi
+          .put({
+            dealCancellationUpdate: tfmDealCancellation,
+            auditDetails: generateTfmAuditDetails(tfmUserId),
+          })
+          .to(`/v1/tfm/deals/${dealId}/cancellation`);
+
+        await testApi
+          .post({ cancellation: tfmDealCancellation, auditDetails: generateTfmAuditDetails(tfmUserId) })
+          .to(`/v1/tfm/deals/${dealId}/cancellation/submit`);
```

Comment:
This does feel like quite a lot to have in a beforeEach & feels a bit tempramental to change. I quite like this actually submitting the deal to make it as close to reality as possible, but since its an api-test I could get rid of the two cancellation calls & just set the stage manually. ie
```suggestion
await testApi
          .put({
            dealType: DEALS.DEAL_TYPE.BSS_EWCS,
            dealId,
            auditDetails: generatePortalAuditDetails(MOCK_PORTAL_USER._id),
          })
          .to('/v1/tfm/deals/submit');

        await testApi
          .put({
            dealUpdate: {
              tfm: {
                dateReceived: '23-09-2024',
                dateReceivedTimestamp: 1727085149,
                parties: {},
                activities: [],
                product: DEALS.DEAL_TYPE.BSS_EWCS,
                stage: TFM_DEAL_STAGE.CANCELLED,
                exporterCreditRating: 'Acceptable (B+)',
                lastUpdated: 1727085149571,
                lossGivenDefault: 50,
                probabilityOfDefault: 12,
              },
            },
            auditDetails: generatePortalAuditDetails(MOCK_PORTAL_USER._id),
          })
          .to(`/v1/tfm/deals/${dealId}`);

        const tfmDealCancellation = { reason: '', effectiveFrom: new Date().valueOf(), bankRequestDate: new Date().valueOf() };
```